content/geometry/CircleTangents.h

```diff
@@ -0,0 +1,32 @@
+/**
+ * Author: Victor Lecomte, chilli
+ * Date: 2019-10-31
+ * License: CC0
+ * Source: https://vlecomte.github.io/cp-geo.pdf
+ * Description: Returns the external tangents of two circles, where P1 and P2
+ * are the tangency points of circle 1 and 2 respectively. Can return 0, 1, or
+ * 2 tangents. If the circles have no tangents/are identical, returns 0 tangents. If there is
+ * only 1 tangent, then the circles are tangent to each other at some point P,
+ * and the tangent line can be found as the perpendicular line c1 -> P passing
+ * through P. Internal tangents can be found by negating r2. To find the
+ * tangents of a circle with a point c2 set r2 to 0.
```

Comment:
I tried to do some wordsmithing here. How does the following sound to you?
```
Description: Finds the external tangents of two circles, or internal if r2 is
negated. Can return 0, 1, or 2 tangents – 0 if one circle contains the other (or
overlaps it, in the internal case); 1 if the circles are tangent to each other (in
which case .first = .second and the tangent line is perpendicular to the line
between the centers). To find the tangents of a circle with a point c2 set r2
to 0. .first and .second give the tangency points at circle 1 and 2 respectively.
```

It tries to fix a few points:
* `can be found as the perpendicular line c1 -> P` needs "of the line", but also isn't totally correct if c1 = P, so I changed this to the perpendicular of the line c1 - c2
* clarify when the 0/1 tangent cases occur

Reply:
Seems a lot better. I might move `To find the tangents of a circle with a point c2 set r2
to 0.` to the end. Also, I want to clarify that if there are infinite tangents, it also returns 0.